3.10 Release Notes

ahardin-rh opened the pull request:
Preview build: http://file.rdu.redhat.com/~ahardin/06272018/3-10-release-notes/release_notes/ocp_3_10_release_notes.html

ahardin-rh commented on dev_guide/topics/about_device_manager.adoc:
```diff
@@ -1,20 +1,5 @@
 
-[IMPORTANT]
-====
-Device Manager is a Technology Preview feature.
```
@vikaschoudhary16 Can you please confirm that Device Manager is coming out of Tech Preview?

ahardin-rh replied:
Confirmed with Jeremy that it's GA in 3.10

ahardin-rh commented on release_notes/ocp_3_10_release_notes.adoc:
```diff
@@ -235,73 +319,73 @@ features marked *GA* indicate _General Availability_.
+|TP
 |TP
 
 |xref:ocp-310-hugepages[Huge Pages]
```
@derekwaynecarr Are Huge Pages being promoted to supported in 3.10, or will they remain in Tech Preview?

ahardin-rh replied:
Oh, I see now that they are: https://github.com/openshift/openshift-docs/pull/8706

wjiangjay commented on release_notes/ocp_3_10_release_notes.adoc:
```diff
@@ -54,33 +54,524 @@ documentation.
+limitations. In particular, they are limited to 2Mi page sizes. THP can lead to
+performance degradation on nodes with high memory utilization or fragmentation
+due to defragmenting efforts of THP, which can lock memory pages. For this
+reason, some applications may be designed to (or recommend) usage of
```
Does this sentence `designed to (or recommend) usage of pre-allocated huge pages` have syntax error?

xingxingxia commented on release_notes/ocp_3_10_release_notes.adoc:
```diff
@@ -54,33 +54,524 @@ documentation.
+Each of the system services, API, controllers, and etcd, used to run as system
+services on the master. These services now run on static pods in the cluster. As
+a result, there are new commands to restart these services: `master-restart
+api`, `master-restart controllers`, and m`aster-restart etcd`. To view log
```
> m\`aster-restart etcd`

Wrong \` place

ganhuang commented on release_notes/ocp_3_10_release_notes.adoc:
```diff
@@ -54,33 +54,524 @@ documentation.
+* The containerized mode for {product-title} is no longer supported (where
+ components run in docker containers) and the 3.10 upgrade will require you to
+ move to RPMs or system containers for the kubelet by setting
+ `openshift_use_system_containers=true`.
```
`openshift_use_system_containers` is removed in https://github.com/openshift/openshift-ansible/pull/8329.

The changes will be done automatically according to the OS during the upgrade:

- Atomic Host: Node service run as system container
- RHEL: Node service run as rpm

ganhuang commented on release_notes/ocp_3_10_release_notes.adoc:
```diff
@@ -54,33 +54,524 @@ documentation.
+master. The 3.10 upgrade, unless you set `openshift_node_bootstrap=false` in your
+inventory, will automatically transform your nodes to use that mode.  This
+enables certificate rotation and centralized certificate management by default. Use `oc get
+csr` to see pending certificates.
```
`openshift_node_bootstrap` is also removed, please see https://bugzilla.redhat.com/show_bug.cgi?id=1567783.

All nodes must be bootstrapped.

liangxia commented on release_notes/ocp_3_10_release_notes.adoc:
```diff
@@ -54,33 +54,550 @@ documentation.
+their current application.
+
+An external provisioner is used to access the EBS, GCE pDisk, and HostPath, and
+Cinder snapshotting API. This Technology Preview feature has tested EBS and
```
I believe Cinder is not supported at the moment. And QE have tested EBS and GCE pDisk, but not HostPath.

You can got the details from https://trello.com/c/owCRnP8I/521-13-snapshot-implement-the-out-of-tree-snapshotting-controller

liangxia commented on release_notes/ocp_3_10_release_notes.adoc:
```diff
@@ -54,33 +54,550 @@ documentation.
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: snapshot-demo
```
Better use a different name with the snapshot name as it can confuse end user.

ahardin-rh replied:
@liangxia I was previously instructed by @sferich888 to change it to match. Eric, can I change it back?

qinpingli commented on release_notes/ocp_3_10_release_notes.adoc:
```diff
@@ -54,33 +54,550 @@ documentation.
+workloads.
+
+You can expand persistent volume claims online from {product-title} for CNS
+glusterFS, Cinder, and GCE PD.
```
Cinder pv does not support online expanding.

zhouying7780 commented:
For this bug: https://bugzilla.redhat.com/show_bug.cgi?id=1593635, we need the 'm3.large' instance in AWS,  I think we should mention in the release doc .

ahardin-rh commented:
@vikram-redhat I applied all of the latest edits, added in all suggested information, and added all of the latest bug descriptions to date. Although this is still a work in progress, I am going to merge this now so that the team can build on this while I'm out on PTO.

ahardin-rh commented:
/cherrypick enterprise-3.10
